### PR TITLE
Sep12 docfix

### PR DESCRIPTION
**Pull request by EitanSchwartz**
All changes are extracted from this doc:
[](https://docs.google.com/document/d/1cmZ4axh0OPRDrjfEjhw1EJcGlKUcHUmeEEjb9e0bBHI/edit?usp=sharing)

### CODE REVIEW

**Review by orenktaboola — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -265,109 +312,119 @@ LocalBroadcastManager.getInstance(getActivity()).registerReceiver(
...
+1. Open the project as you would any other Android project.
+2. Taboola is optimized to working with Android Studio but other IDEs should work as well.
+
+##### 3.3 Example App As Troubleshooting Helper:
+In case you encounter some issues while integrating the SDK into your app, try to recreate the scenario within the example app. This might help to isolate the problems. For more help, you would be able to send the example app with your recreated issue to Taboola's support.
+
+## 4. SDK Reference - TaboolaWidget
+## `Notice: Dear reviewer, my definitions here might be incorrect/innacurate`
+##### `public TaboolaWidget setPublisher(String publisher)`
+**Mandatory**. Sets the `publisher` (The name of your application in Taboola Admin)
```
Comment (bkoifman) [inline README.md]: Publisher - Given from Taboola's representative this is the publisher name (you account name in Taboola),

File: README.md [hunk 2]
```diff
@@ -265,109 +312,119 @@ LocalBroadcastManager.getInstance(getActivity()).registerReceiver(
...
+
+##### `public TaboolaWidget setMode(String mode)`
+**Mandatory**. Sets the widget display `mode` (The name of your application in Taboola Admin)
```
Comment (bkoifman) [inline README.md]: Mode - Taboola UI mode, again given by Taboola representatives.

File: README.md [hunk 3]
```diff
@@ -265,109 +312,119 @@ LocalBroadcastManager.getInstance(getActivity()).registerReceiver(
...
+
+##### `public TaboolaWidget setPlacement(String placement)`
+**Mandatory**. Sets the widget `placement` (can also be set via XML as `placement`)
```
Comment (bkoifman) [inline README.md]: Placement - As a part of Taboola's integration were defining different blocks of UI in your app to be displayed in different placements, each of those will hold a meaningful name .
Taboola's configuration relies on the placement name to serve correct content & UI correctly.
Again sent by Taboola representatives.

File: README.md [hunk 4]
```diff
@@ -265,109 +312,119 @@ LocalBroadcastManager.getInstance(getActivity()).registerReceiver(
...
+
+##### `public TaboolaWidget setPageType(String pageType)`
+**Mandatory**. (Can also be set via XML as `page_type`)
+
+##### `public TaboolaWidget setPageUrl(String pageUrl)`
+**Mandatory**. (Can also be set via XML as `url`)
```
Comment (bkoifman) [inline README.md]: The URL of the current page, Taboola uses it to differ the different client pages.

File: README.md [hunk 5]
```diff
@@ -265,109 +312,119 @@ LocalBroadcastManager.getInstance(getActivity()).registerReceiver(
...
+**Mandatory**. Sets the `publisher` (The name of your application in Taboola Admin)
+
+##### `public TaboolaWidget setMode(String mode)`
+**Mandatory**. Sets the widget display `mode` (The name of your application in Taboola Admin)
+
+##### `public TaboolaWidget setPlacement(String placement)`
+**Mandatory**. Sets the widget `placement` (can also be set via XML as `placement`)
+
+##### `public TaboolaWidget setPageType(String pageType)`
+**Mandatory**. (Can also be set via XML as `page_type`)
```
Comment (bkoifman) [inline README.md]: Given from Taboola's representatives, those are defining different type of pages as source and target and by those Taboola distinguish the recommendations required,

File: README.md (hunk 2 above)
Comment (bkoifman) [reply README.md]: UI Layout

Comment (bkoifman) [reply README.md]: The arrangment of content items in your Placement

File: README.md (hunk 3 above)
Comment (bkoifman) [reply README.md]: Place before the mode part(as were explaining the mode with the placement)

File: README.md (hunk 5 above)
Comment (bkoifman) [reply README.md]: This is correct for both source and target types.

t - text, v - video, s - search, p - photo, c - category, h - homepage, o - other, m - mix (a mix of item types, not a homogenous recommendation list)

File: README.md (hunk 4 above)
Comment (bkoifman) [reply README.md]: Taboola holds reference to each of your app activities as pages,
Taboola requires you to differ the app activities using unique URL for each.

**Pull request merged**